src/mbgl/tile/geometry_tile.cpp

```diff
@@ -133,7 +141,9 @@ void GeometryTile::onLayout(LayoutResult result) {
 void GeometryTile::onPlacement(PlacementResult result) {
     loaded = true;
     renderable = true;
-    if (result.correlationID == correlationID) {
+    if (result.layoutCorrelationID == lastLayoutCorrelationID
```

Comment:
> This placement result is using the same layout as the last layout result
The last layout result was for the most recent layout request
This placement result is for the most recent placement request
Then mark this tile no longer pending. Is that right?

:100: 

> The logic before this change was, "if this placement is the first placement to happen in response to the last call to any of setData/setLayers/setPlacementConfig, then mark this tile as no longer pending". In order for that logic to work, the critical assumption was that while the worker might receive onImagesAvailable or onGlyphsAvailable, it would only do placement if it had everything it needed to proceed to completion.

Precisely - and as exemplified in `API.RecycleMapUpdateImages` that is an assumption we cannot rely.